### PR TITLE
IP Grades v2 + Auth

### PR DESCRIPTION
**Pull request by skorpiom**
Hey,

changes in the test grades routes:
get: /grades/:userID
post: /grades/:userID
patch: /grades/:id
delete: /grades/:id

so right now it doesn't create seperate collection, just adds grades to user grades array
that routes are accessible just for admins and mentors

mentor can just use that routes to modify users grades that are in the same team
mentor can't modify users grades from other teams

### CODE REVIEW

File: Api/Src/Services/GradeService.ts [hunk 1]
```diff
@@ -1,26 +1,47 @@
...
+
+        await this.userService.updateUser(user._id, user);
+        return grade;
     }
-    findGrade = async (id: mongoose.Types.ObjectId) => {
-        return this.repository.getById(id);
+
+    findGrades = async (req: express.Request) => {
+        const id = new mongoose.Types.ObjectId(req.params.userID);
+        const user = await this.userService.findUserById(id);
```
Comment (robert-czwartosz) [inline Api/Src/Services/GradeService.ts:31]: if(!user) throw new Error("User doesn't exists");

**Review by robert-czwartosz — COMMENTED** [review]
Generally, api works good, but there are 2 issues

File: Api/Src/Controllers/GradeController.ts [hunk 2]
```diff
@@ -48,13 +34,9 @@ export default class GradeController {
...
         try {
-            const id = new mongoose.Types.ObjectId(req.params.id);
-            const grade = new GradeSchema(req.body);
-            grade._id = id;
-            await grade.validate();
-            const updatedGrade = await this.service.updateGrade(id, grade);
-            return updatedGrade ?
-                res.status(201).json(updatedGrade)
+            const result = await this.service.updateGrade(req);
+            return result ?
```
[the remark this answers is not in the document]
Comment (skorpiom) [reply Api/Src/Controllers/GradeController.ts]: done

File: Api/Src/Services/GradeService.ts (hunk 1 above)
Comment (skorpiom) [reply Api/Src/Services/GradeService.ts:31]: added, thanks ;)

**Review by robert-czwartosz — APPROVED** [review]

**Review by fszwedo — APPROVED** [review]
looks fine :)

**Pull request merged**